### PR TITLE
Update free-programming-books-langs.md

### PR DESCRIPTION
**Pull request by vineetsingh0809**
## What does this PR do?
Add resource(s)

### Description
Here we added books on python practical programming.

### Why is this valuable (or not)?
Because we can prepare practicals with these books.

### How do we know it's really free?
I have attached the free link to the ebook.

### For book lists, is it a book? For course lists, is it a course? etc.
Book

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1899,6 +1900,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Natural Language Processing with Python](http://www.nltk.org/book/) (3.x)
 * [Non-Programmer's Tutorial for Python 3](https://en.wikibooks.org/wiki/Non-Programmer%27s_Tutorial_for_Python_3) - Wikibooks (3.3)
   * [Non-Programmer's Tutorial for Python 2.6](https://en.wikibooks.org/wiki/Non-Programmer%27s_Tutorial_for_Python_2.6) - Wikibooks (2.6)
+* [Obey the Testing Goat!](https://www.obeythetestinggoat.com/pages/book.html) - Harry Percival (HTML, PDF, Exercises sources)
```
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-langs.md]: Yes. you are right.

Remove this line and complete the other:

https://github.com/EbookFoundation/free-programming-books/blob/8b725705715ab48354c6340bd56a71d20859ab15/books/free-programming-books-langs.md#L1940

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -1705,6 +1705,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [PHP Apprentice](https://phpapprentice.com) - Andrew Davis *(:construction: in process)*
 * [PHP Best Practices](https://phpbestpractices.org)
 * [PHP Documentor - Documentation](https://docs.phpdoc.org) - phpdoc.org
+* [PHP Essentials](https://www.techotopia.com/index.php/PHP_Essentials) - Krzysztof Kowalczyk(HTML, PDF, Exercises sources)
```
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-langs.md:1708]: If author is really Krzysztof Kowalczyk, the right site would be: https://www.programming-books.io/essential/php/ , another StackOverflow documentation rip-off

**Comment by vineetsingh0809** [comment]
Hey @eshellman Sir, I made all the changes that you told me. Check it again

File: books/free-programming-books-langs.md [hunk 3]
```diff
@@ -798,6 +798,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Introduction to Git and Github](https://launchschool.com/books/git) - Launch School
 * [Introduction to Git and Github - Tutorial](http://cse.unl.edu/~cbourke/gitTutorial.pdf) - Dr. Chris Bourke (PDF)
 * [Introduction to Git and GitHub eBook](https://github.com/bobbyiliev/introduction-to-git-and-github-ebook) - Bobby Iliev (Markdown, PDF)
+* [Learn Enough Git to Be Dangerous](https://www.learnenough.com/git-tutorial/getting_started) - Michael Hartl(HTML, PDF, Exercises sources)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:801]: Freemium: only free 2 chapters of

File: books/free-programming-books-langs.md [hunk 4]
```diff
@@ -1937,7 +1940,6 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [SICP in Python](http://www-inst.eecs.berkeley.edu/~cs61a/sp12/book/) (3.2)
 * [Slither into Python: An introduction to Python for beginners](https://web.archive.org/web/20210411065902/https://www.slitherintopython.com/) (3.7) *(:card_file_box: archived)*
 * [Supporting Python 3: An In-Depth Guide](http://python3porting.com) (2.6 - 2.x & 3.1 - 3.x)
-* [Test-Driven Web Development with Python](http://www.obeythetestinggoat.com/pages/book.html) - Harry Percival (3.3 - 3.x)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:1940]: Why remove this? The wrong was the other!

File: books/free-programming-books-langs.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-langs.md:1904]: Removing patch to complete https://github.com/EbookFoundation/free-programming-books/pull/6496/files#r739629120 suggestion
```suggestion
```

File: books/free-programming-books-langs.md [hunk 5]
```diff
@@ -1899,6 +1901,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Natural Language Processing with Python](http://www.nltk.org/book/) (3.x)
 * [Non-Programmer's Tutorial for Python 3](https://en.wikibooks.org/wiki/Non-Programmer%27s_Tutorial_for_Python_3) - Wikibooks (3.3)
   * [Non-Programmer's Tutorial for Python 2.6](https://en.wikibooks.org/wiki/Non-Programmer%27s_Tutorial_for_Python_2.6) - Wikibooks (2.6)
+* [Obey the Testing Goat!](https://www.obeythetestinggoat.com/pages/book.html) - Harry Percival (HTML, PDF, Exercises sources)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:1904]: Removing patch to complete https://github.com/EbookFoundation/free-programming-books/pull/6496#discussion_r739629120 suggestion
```suggestion
```

**Review by davorpa — CHANGES_REQUESTED** [review]

File: books/free-programming-books-langs.md (hunk 4 above)
Comment (davorpa) [reply books/free-programming-books-langs.md:1940]: Not resolved yet https://github.com/EbookFoundation/free-programming-books/pull/6496/files#r739629120

Comment (davorpa) [reply books/free-programming-books-langs.md:1940]: https://github.com/EbookFoundation/free-programming-books/pull/6496/files#r739629120

**Review by davorpa — CHANGES_REQUESTED** [review]

File: books/free-programming-books-langs.md [hunk 6]
```diff
@@ -1937,7 +1939,6 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [SICP in Python](http://www-inst.eecs.berkeley.edu/~cs61a/sp12/book/) (3.2)
 * [Slither into Python: An introduction to Python for beginners](https://web.archive.org/web/20210411065902/https://www.slitherintopython.com/) (3.7) *(:card_file_box: archived)*
 * [Supporting Python 3: An In-Depth Guide](http://python3porting.com) (2.6 - 2.x & 3.1 - 3.x)
-* [Test-Driven Web Development with Python](http://www.obeythetestinggoat.com/pages/book.html) - Harry Percival (3.3 - 3.x)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:1940]: Not resolved yet https://github.com/EbookFoundation/free-programming-books/pull/6496#discussion_r739629120

File: books/free-programming-books-langs.md (hunk 3 above)
Comment (davorpa) [inline books/free-programming-books-langs.md:801]: Only free 38%...

Apply following delete patch here:
```suggestion
```

File: books/free-programming-books-langs.md [hunk 7]
```diff
@@ -1937,7 +1939,6 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [SICP in Python](http://www-inst.eecs.berkeley.edu/~cs61a/sp12/book/) (3.2)
 * [Slither into Python: An introduction to Python for beginners](https://web.archive.org/web/20210411065902/https://www.slitherintopython.com/) (3.7) *(:card_file_box: archived)*
 * [Supporting Python 3: An In-Depth Guide](http://python3porting.com) (2.6 - 2.x & 3.1 - 3.x)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:1941]: **PLEASE! Apply following patch as-is** to recover deleted resource at next line (L1940)
```suggestion
* [Supporting Python 3: An In-Depth Guide](http://python3porting.com) (2.6 - 2.x & 3.1 - 3.x)
* [Test-Driven Web Development with Python: Obey the Testing Goat! using Django, Selenium and Javascript](http://www.obeythetestinggoat.com/pages/book.html) - Harry J.W. Percival (HTML) *(3.3 - 3.x)*
```
It also resolve ALL previous suggestions proposed before about it: `Test-Driven Web Development with Python: Obey the Testing Goat! using Django, Selenium and Javascript`

**Review by davorpa — CHANGES_REQUESTED** [review]
**PLEASE! Apply following patches as-is**

It will resolve all your summited dups PRs.

**Comment by vineetsingh0809** [comment]
Hey @davorpa, Firstly, thanks a lot for this kind of help that you did and guided me in the right way. Thank you so much.
BTW I made the changes you told me to do...So please check it again and merge it if this PR is eligible for merge...

**Comment by vineetsingh0809** [comment]
And please review this PR as well ...... #6547
please.....please

**Review by davorpa — APPROVED** [review]

**Pull request merged**